lib/dry/transaction.rb

```diff
@@ -75,5 +75,9 @@ def initialize(steps, matcher)
       @steps = steps
       @matcher = matcher
     end
+
+    def with_step_arguments(new_steps)
+      self.class.new(new_steps, matcher)
+    end
```

Comment:
Can we move this `self.class.new(new_steps, matcher)` inside your other `#step_args` method? I'm not sure why we would want a second method added to our API just for passing another set of pre-built steps.